bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,269 @@
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.vektiva.internal.config.VektivaSmarwiConfiguration;
+import org.openhab.binding.vektiva.internal.net.VektivaSmarwiSocket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.openhab.binding.vektiva.internal.VektivaBindingConstants.*;
+
+/**
+ * The {@link VektivaSmarwiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class VektivaSmarwiHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VektivaSmarwiHandler.class);
+
+    private VektivaSmarwiConfiguration config = new VektivaSmarwiConfiguration();
+
+    private final HttpClient httpClient;
+
+    private final WebSocketClient webSocketClient;
+
+    private @Nullable Session session;
+
+    private @Nullable ScheduledFuture<?> future = null;
+
+    private int lastPosition = -1;
+
+    public VektivaSmarwiHandler(Thing thing, HttpClient httpClient, WebSocketClient webSocketClient) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.webSocketClient = webSocketClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (CHANNEL_STATUS.equals(channelUID.getId()) && command instanceof RefreshType) {
+            checkStatus();
+            return;
+        }
+
+        if (CHANNEL_CONTROL.equals(channelUID.getId())) {
+            logger.debug("Received command: {}", command);
+            String cmd = getSmarwiCommand(command);
+            if (COMMAND_OPEN.equals(cmd) || COMMAND_CLOSE.equals(cmd) || COMMAND_STOP.equals(cmd)) {
+                if (RESPONSE_OK.equals(sendCommand(cmd))) {
+                    lastPosition = COMMAND_OPEN.equals(cmd) ? 0 : 100;
```

Comment:
What if cmd is `COMMAND_STOP`? Is `100` the correct position?

Reply:
yes, if COMMAND_STOP is received, the device is unable to report its state, so its closed and lastPosition is 100;